models/event.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+const Build = require('./build');
+const Joi = require('joi');
+const mutate = require('../lib/mutate');
+const Pipeline = require('./pipeline');
+const Scm = require('../core/scm');
+const Workflow = require('../config/workflow');
+
+const buildId = Joi.reach(Build.base, 'id');
+
+const MODEL = {
```

Comment:
Initially, I wanted to defer with saving the overall status in the event model, just because I'm not sure we need it quite yet. It would be convenient. 

Some of the CD solutions we looked at earlier this year gave us some options on "what is deployment success" (e.g., If half of the machines are deployed to, consider it a success). Maintaining the status in the `event` model would enable that possibility. 

I'll add it in the next update as well.